platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/state/DefaultStateManagementComponent.java

```diff
@@ -88,16 +89,18 @@ public DefaultStateManagementComponent(
         this.signedStateSentinel = new SignedStateSentinel(platformContext, threadManager, Time.getCurrent());
         this.stateSigner = Objects.requireNonNull(stateSigner);
         this.sigCollector = Objects.requireNonNull(sigCollector);
-
-        hashLogger =
-                new HashLogger(threadManager, platformContext.getConfiguration().getConfigData(StateConfig.class));
+        this.hashLogger = Objects.requireNonNull(hashLogger);
 
         signedStateHasher = new SignedStateHasher(signedStateMetrics, fatalErrorConsumer);
     }
 
     private void logHashes(final SignedState signedState) {
         if (signedState.getState().getHash() != null) {
-            hashLogger.logHashes(signedState);
+            ReservedSignedState rss = signedState.reserve("logging hash state");
+            boolean offered = hashLogger.test(rss);
```

Comment:
The task is *always* being offered, whether or not it's actually accepted. Therefore, I think a better name for this variable would be `offerAccepted`.